core/string/ustring.cpp

```diff
@@ -3772,9 +3772,9 @@ String String::uri_encode() const {
 		} else {
 			char h_Val[3];
 #if defined(__GNUC__) || defined(_MSC_VER)
-			snprintf(h_Val, 3, "%hhX", ord);
+			snprintf(h_Val, 3, "%02X", ord);
 #else
-			sprintf(h_Val, "%hhX", ord);
+			sprintf(h_Val, "%02X", ord);
```

Comment:
This seems okay, but shouldn't we keep the `hh`? i.e.: `"%02hhX"`?
@RandomShaper opinions?

Reply:
I think we should. Ohterwise, if `ord < 0`, the formatting would take more than two characters despite the `02`: https://onlinegdb.com/SJTQ1Bnru